cmake/CUDAConfig.cmake

```diff
@@ -0,0 +1,10 @@
+
+# CUDA Options
+if(CUDA_VERSION_STRING VERSION_GREATER 10)
+  # 2829 annotation on a defaulted function is ignored
+  # 3057 annotation is ignored on a function that is explicitly defaulted
+  # 2929 annotation is ignored on a function that is explicitly defaulted
+  target_compile_options(${TARGET_NAME} PRIVATE $<$<COMPILE_LANGUAGE:CUDA>:-Xcudafe="--diag_suppress=3057,2929">)
+else()
+  target_compile_options(${TARGET_NAME} PRIVATE $<$<COMPILE_LANGUAGE:CUDA>:-Xcudafe="--diag_suppress=2829">)
+endif()
```

Comment:
Having this in a separate file is great, but it doesn't get used. I'd wrap it into a function (called e.g., `apply_cuda_target_options`) that accepts `TARGET_NAME` as a parameter, include this file from the top-level `CMakeLists.txt` file, then call the function where you currently have this code duplicated.